File: ios/Stride/Theme/Theme.swift
import SwiftUI

/// Stride cinematic theme — midnight navy + jewel-tone accents.
enum Theme {
    static let bg = Color(red: 0.024, green: 0.027, blue: 0.051)
    static let bgElev = Color(red: 0.047, green: 0.059, blue: 0.102)
    static let card = Color(red: 0.075, green: 0.094, blue: 0.149)
    static let surface = Color(red: 0.106, green: 0.133, blue: 0.200)
    static let border = Color(red: 0.141, green: 0.188, blue: 0.286)

    static let text = Color(red: 0.960, green: 0.968, blue: 0.984)
    static let textMuted = Color(red: 0.604, green: 0.639, blue: 0.722)
    static let textDim = Color(red: 0.353, green: 0.388, blue: 0.470)

    static let gold = Color(red: 0.831, green: 0.686, blue: 0.216)
    static let goldBright = Color(red: 0.957, green: 0.816, blue: 0.247)
    static let emerald = Color(red: 0.063, green: 0.725, blue: 0.506)
    static let ruby = Color(red: 0.937, green: 0.267, blue: 0.267)
    static let sapphire = Color(red: 0.231, green: 0.510, blue: 0.965)
}

enum Tier: String, CaseIterable, Codable, Sendable {
    case bronze, silver, gold, platinum

    var label: String { rawValue.capitalized }

    var color: Color {
        switch self {
        case .bronze: Color(red: 0.804, green: 0.498, blue: 0.196)
        case .silver: Color(red: 0.788, green: 0.812, blue: 0.859)
        case .gold: Color(red: 0.831, green: 0.686, blue: 0.216)
        case .platinum: Color(red: 0.490, green: 0.827, blue: 0.988)
        }
    }

    var glow: Color {
        switch self {
        case .bronze: Color(red: 0.902, green: 0.620, blue: 0.353)
        case .silver: Color(red: 0.910, green: 0.929, blue: 0.961)
        case .gold: Color(red: 0.957, green: 0.816, blue: 0.247)
        case .platinum: Color(red: 0.729, green: 0.902, blue: 0.992)
        }
    }

    var gradient: [Color] {
        [color.opacity(0.95), glow]
    }

    /// Random reward range per tier — v3 (raffles-only era): cut ~25% so weekly
    /// coin supply lines up with a £500–£1,500 raffle outlay.
    var rewardRange: ClosedRange<Int> {
        switch self {
        case .bronze: 26...45
        case .silver: 60...105
        case .gold: 135...195
        case .platinum: 270...390
        }
    }

    var xpRange: ClosedRange<Int> {
        switch self {
        case .bronze: 25...40
        case .silver: 55...80
        case .gold: 120...180
        case .platinum: 240...320
        }
    }
}

File: expo/constants/theme.ts
/**
 * Stride Quest theme — cinematic dark "treasure hunt at night" palette.
 * Deep midnight navy backgrounds, gold + emerald jewel-tone accents.
 */
export const theme = {
  // base
  bg: "#06070D",
  bgElev: "#0C0F1A",
  bgCard: "#131826",
  surface: "#1B2233",
  surfaceHi: "#222B40",
  border: "#243049",
  borderSoft: "#1A2236",
  divider: "#1B2233",

  // text
  text: "#F5F7FB",
  textMuted: "#9AA3B8",
  textDim: "#5A6378",

  // brand
  gold: "#D4AF37",
  goldBright: "#F4D03F",
  goldDim: "#7A6420",
  emerald: "#10B981",
  emeraldBright: "#34D399",
  emeraldDim: "#0B6F4C",
  ruby: "#EF4444",
  sapphire: "#3B82F6",

  // map
  thames: "#0E3A4A",
  thamesGlow: "#1E6B82",
  thamesHi: "#2E92AE",
  park: "#0E2A1E",
  parkLight: "#163C2A",
  parkGlow: "#1F5C40",
  road: "#1A2236",
  roadHi: "#243049",
  mapGrad1: "#070A14",
  mapGrad2: "#0C1422",

  // tier colors
  tier: {
    bronze: "#CD7F32",
    bronzeBright: "#E69E5A",
    silver: "#C9CFDB",
    silverBright: "#E8EDF5",
    gold: "#D4AF37",
    goldBright: "#F4D03F",
    platinum: "#7DD3FC",
    platinumBright: "#BAE6FD",
  } as const,

  // layout
  radius: {
    sm: 8,
    md: 12,
    lg: 16,
    xl: 22,
    pill: 999,
  },
  spacing: {
    xs: 4,
    sm: 8,
    md: 12,
    lg: 16,
    xl: 24,
    xxl: 32,
  },
} as const;

export type Tier = "bronze" | "silver" | "gold" | "platinum";

export const tierColor = (t: Tier): string => theme.tier[t];
export const tierGlow = (t: Tier): string => {
  switch (t) {
    case "bronze":
      return theme.tier.bronzeBright;
    case "silver":
      return theme.tier.silverBright;
    case "gold":
      return theme.tier.goldBright;
    case "platinum":
      return theme.tier.platinumBright;
  }
};

export const tierLabel = (t: Tier): string =>
  t.charAt(0).toUpperCase() + t.slice(1);
